cmdLine/src/main/scala/topgun/cmdline/ClassLoaderInfo.scala

```diff
@@ -0,0 +1,34 @@
+package topgun.cmdline
+
+
+import java.io.InputStream
+
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.tree.ClassNode
+
+import scala.collection.mutable
+
+class ClassLoaderInfo() {
+  var allClasses = new mutable.HashMap[String, ClassInfo]()
+
+  def lookup(clazzName: String, clazzLoader: ClassLoader): ClassInfo = {
+    val resourceName = clazzName.replace(".", "/") + ".class"
+    allClasses.getOrElseUpdate(clazzName, buildClassInfo(resourceName, clazzLoader)): ClassInfo
+  }
```

Comment:
```suggestion
  def lookup(className: String, classLoader: ClassLoader): ClassInfo = {
    allClasses.getOrElseUpdate(className, buildClassInfo(className, classLoader)): ClassInfo
  }
 def buildClassInfo(className: String, clazzLoader: ClassLoader): ClassInfo = {
   val resourceName = className.replace(".", "/") + ".class"
 ```

Reply:
okay, will make those changes.
Thanks

Reply:
Placing the resourceName value in the buildClassInfo functions yields a "wrong forward reference" error.
Since that given name is not known at compile time. 
Hence we would probably have to keep the resourceName parameter to the"buildClassInfo" function. 

looking for other ways around it.